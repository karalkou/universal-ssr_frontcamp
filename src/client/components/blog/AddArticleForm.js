import React, { Component } from 'react';
// import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import styled from 'styled-components';
import bemto from 'bemto-components';
import { addArticle } from '../../ducks/articles';

class AddArticleForm extends Component {
    state = {
        title: '',
        body: ''
    };

    handleChange = field => ev => {
        this.setState({
            [field]: ev.target.value
        });
    };

    handleSubmit = ev => {
        ev.preventDefault();
        this.props.addArticle(this.state);

        this.setState({
            title: '',
            body: '',
        })
    };

    render() {
        return (
            <form onSubmit={this.handleSubmit}>
                <div>
                    <input value={this.state.title} onChange={this.handleChange('title')}/>
                </div>
                <div>
                    <textarea value={this.state.body} onChange={this.handleChange('body')}/>
                </div>
                <div>
                    <input type="submit" />
                </div>
            </form>
        );
    }
}

AddArticleForm.propTypes = {};
AddArticleForm.defaultProps = {};

export default connect(
    null,
    { addArticle }
)(AddArticleForm);

/* styles */
var AddArticleFormStyled = styled(bemto({}))``;

