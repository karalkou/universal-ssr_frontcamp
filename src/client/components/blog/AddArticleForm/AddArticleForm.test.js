import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
const store = configureMockStore()({});
import AddArticleForm from './AddArticleForm';

describe('AddArticleForm Component tests', () => {
    it('AddArticleForm match snapshot', () => {
        const component = renderer.create(
            <Provider store={store}>
                <AddArticleForm />
            </Provider>
        );
        const tree = component.toJSON();

        expect(tree).toMatchSnapshot();
    });
});
