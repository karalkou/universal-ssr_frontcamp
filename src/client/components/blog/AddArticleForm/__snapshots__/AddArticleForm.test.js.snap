// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddArticleForm Component tests AddArticleForm match snapshot 1`] = `
<div
  className="sc-bdVaJa kUmsbz"
>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <div>
      <input
        className="field"
        onChange={[Function]}
        value=""
      />
    </div>
    <div>
      <textarea
        className="field"
        onChange={[Function]}
        value=""
      />
    </div>
    <div>
      <input
        type="submit"
      />
    </div>
  </form>
</div>
`;
